crates/turborepo-lib/src/task_graph/visitor.rs

```diff
@@ -1001,15 +1002,33 @@ impl ExecContext {
                     .can_cache(&self.task_hash, &self.task_id_for_display)
                     .unwrap_or(true)
                 {
-                    if let Err(e) = self.task_cache.save_outputs(task_duration, telemetry).await {
-                        error!("error caching output: {e}");
-                        return Err(e.into());
-                    } else {
-                        // If no errors, update hash tracker with expanded outputs
-                        self.hash_tracker.insert_expanded_outputs(
-                            self.task_id.clone(),
-                            self.task_cache.expanded_outputs().to_vec(),
-                        );
+                    let result = self.task_cache.save_outputs(task_duration, telemetry).await;
+                    match result {
+                        Ok(status) => {
+                            match status {
+                                OutputSaveResult::SavedEmpty => {
+                                    prefixed_ui.warn(format!(
+                                        "warning: no files were found that match the configured \
+                                         outputs - make sure \"outputs\" are correctly defined in \
+                                         your `turbo.json` for {}",
```

Comment:
Do we want to write these into the cached task logs so when restoring this hit from cache you'll see this warning?